emrichen/template.py

```diff
@@ -24,9 +24,8 @@ def render(self, context, format='yaml'):
         return render(enriched, format)
 
     @classmethod
-    def parse(cls, data, format='yaml'):
-        filename = None
-        if hasattr(data, 'name') and data.name:
+    def parse(cls, data, format='yaml', filename=None):
+        if filename is None and hasattr(data, 'name') and data.name:
```

Comment:
... Well, fine, `filename = (filename or getattr(data, 'name', None) or None)` then :)

Reply:
I prefer the current form 😅 the one that has less parentheses wins